apm-agent-plugins/apm-urlconnection-plugin/src/main/java/co/elastic/apm/agent/urlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -0,0 +1,145 @@
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+public abstract class HttpUrlConnectionInstrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Arrays.asList("http-client", "urlconnection");
+    }
+
+    @Override
+    public ElementMatcher<? super NamedElement> getTypeMatcherPreFilter() {
+        return nameContains("URLConnection").or(nameContains("UrlConnection"));
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return hasSuperType(is(HttpURLConnection.class));
+    }
+
+    public static class CreateSpanInstrumentation extends HttpUrlConnectionInstrumentation {
+
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void enter(@Advice.This HttpURLConnection thiz,
+                                 @Advice.FieldValue("connected") boolean connected) {
+            if (tracer == null || tracer.getActive() == null || connected) {
+                return;
+            }
+            final URL url = thiz.getURL();
+            final Span span = HttpClientHelper.startHttpClientSpan(tracer.getActive(), thiz.getRequestMethod(), url.toString(), url.getHost(), "urlconnection");
+            if (span != null) {
+                span.setOriginator(thiz);
+                if (thiz.getRequestProperty(TraceContext.TRACE_PARENT_HEADER) == null) {
+                    thiz.addRequestProperty(TraceContext.TRACE_PARENT_HEADER, span.getTraceContext().getOutgoingTraceParentHeader().toString());
+                }
+            }
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+        public static void exit(@Advice.This HttpURLConnection thiz,
+                                @Nullable @Advice.Thrown Throwable t,
+                                @Advice.FieldValue("responseCode") int responseCode) {
+            if (tracer == null) {
+                return;
+            }
+            // we can't use local variables as a span might be started in HttpUrlConnection#connect
+            // but it is always ended in HttpUrlConnection#getInputStream
+            final TraceContextHolder<?> active = tracer.getActive();
+            if (active instanceof Span) {
+                final Span span = (Span) active;
+                // makes sure this is actually the span for this HttpUrlConnection and not some random parent span
+                if (span.isOriginatedBy(thiz)) {
+                    if (responseCode != -1) {
+                        // if the response code is set, the connection has been established via getOutputStream
+                        // if the response code is unset even after getOutputStream has been called, there will be an exception
+                        span.getContext().getHttp().withStatusCode(responseCode);
+                        span.captureException(t).deactivate().end();
```

Comment:
maybe set the `Originator` to `null` once we don't need it? Just in case this span is going to wait a long time in the disruptor queue, make it eligible for gc